pkg/util/files/creator.go

```diff
@@ -0,0 +1,75 @@
+package files
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/spf13/afero"
+)
+
+//FileCreator is an interface to encapsulate the file creation process. Is has 2 implementations
+//in memory and one that uses files on disk.
+type FileCreator interface {
+	CreateFolder(path string) (fullpath string, err error)
+	CreateFile(byteArray []byte, path string, name string, fileType string) (cleanName string, err error)
+}
+
+// fileCreatorImpl implements interface FileCreator using disk as storage
+type fileCreator struct {
+	fileManager afero.Fs
+}
+
+//NewDiskFileCreator creates a new FileCreator instance
+func NewDiskFileCreator() FileCreator {
+	el := &fileCreator{}
+	el.fileManager = afero.NewOsFs()
+	return el
+}
+
+//CreateFolder creates a folder in the specified path
+func (a *fileCreator) CreateFolder(path string) (fullpath string, err error) {
+	//path should be sanitized
+	if _, err := os.Stat(path); os.IsNotExist(err) {
```

Comment:
Having taken a quick look at the library it does indeed seem like it might be a good way to replace the self-made abstraction layer in files.go

At a glance it seems to have some nice features in terms of testing. 

@didiladi what's your take?

From my side: If we intend to use it more thoroughly it would seem reasonable to add it.

Reply:
It makes sense to simplify/replace the abstraction layer in `files.go` if all use-cases like, e.g. in-memory filesystem are also supported in `afero`. But that would need some more investigation. As we don't want to block @diegorqc's PR here, for now, I'd say, its good as it is - let's revisit that in a followup ticket. :) Everyone ok with that?

Reply:
Sound good to me :+1:

Reply:
yes ``afero`` is a good library to mock the filesystem, it provides some good examples about how to mock everything and make it unit testable. I have modified line 34 to use ``afero.Exist``